receiver/k8sclusterreceiver/internal/collection/containers.go

```diff
@@ -111,22 +112,52 @@ func getSpecMetricsForContainer(c corev1.Container) []*metricspb.Metric {
 	} {
 		for k, v := range t.rl {
 			val := v.Value()
-			if k == corev1.ResourceCPU {
-				val = v.MilliValue()
-			}
-
-			metrics = append(metrics,
-				&metricspb.Metric{
-					MetricDescriptor: &metricspb.MetricDescriptor{
-						Name:        fmt.Sprintf("k8s.container.%s_%s", k, t.typ),
-						Description: t.description,
-						Type:        metricspb.MetricDescriptor_GAUGE_INT64,
+			if featuregate.IsEnabled(reportCPUMetricsAsDoubleFeatureGateID) {
```

Comment:
maybe something like this to avoid duplicated code? 

```go
val = utils.GetInt64TimeSeries(v.Value())
valType := metricspb.MetricDescriptor_GAUGE_INT64
if k == corev1.ResourceCPU {
    if featuregate.IsEnabled(reportCPUMetricsAsDoubleFeatureGateID) {
        valType = metricspb.MetricDescriptor_GAUGE_DOUBLE
        val = utils.GetDoubleSeries(float64(v.MilliValue()) / 1000.0)
    } else {
        val = utils.GetInt64TimeSeries(v.MilliValue())
    }
}
```

Reply:
Sounds good to me.